### PR TITLE
First sweep of untangle patches.

### PR DESCRIPTION
**Pull request by obnoxxx**
Preparations for introducing pending operations.

### CODE REVIEW

**Comment by obnoxxx** [comment]
cleaned-up part of untangle work for volume.

**Comment by obnoxxx** [comment]
@phlogistonjohn wrote:
> LGTM. But of course I wrote most of it. ;-)

small correction: all of it (since I removed the other patches from the top for now) ;-)

File: apps/glusterfs/volume_entry.go [hunk 1]
```diff
@@ -292,88 +261,126 @@ func (v *VolumeEntry) Create(db wdb.DB,
...
+	// Create the bricks on the nodes
+	e = CreateBricks(db, executor, brick_entries)
+	if e != nil {
+		return
 	}
 
-	// Destroy volume on failure
-	defer func() {
-		if e != nil {
-			v.Destroy(db, executor)
```
Comment (raghavendra-talur) [inline apps/glusterfs/volume_entry.go]: This is the v.Destroy I was referring to in the other comment.

File: apps/glusterfs/volume_entry_create.go [hunk 2]
```diff
@@ -37,6 +37,11 @@ func (v *VolumeEntry) createVolume(db wdb.RODB,
 	if _, err := executor.VolumeCreate(host, vr); err != nil {
 		return err
 	}
```
Comment (raghavendra-talur) [inline apps/glusterfs/volume_entry_create.go]: This is possibly where we call v.Destroy.

**Pull request merged**